download/src/lib.rs

```diff
@@ -277,40 +250,56 @@ pub mod reqwest_be {
 
     use super::Event;
     use super::TlsBackend;
+    use super::{Callback, Result};
     use crate::errors::*;
 
     pub fn download(
         url: &Url,
         resume_from: u64,
-        callback: &dyn Fn(Event<'_>) -> Result<()>,
+        callback: Callback<'_>,
         tls: TlsBackend,
     ) -> Result<()> {
-        // Short-circuit reqwest for the "file:" URL scheme
-        if download_from_file_url(url, resume_from, callback)? {
-            return Ok(());
-        }
+        // The file scheme is mostly for use by tests to mock the dist server.
```

Comment:
It is also used by some organisations with mirrors on NFS - that kind of thing. I think it is a mistake to over-emphasis the test usage here - if we truely wanted to make it test only, we'd make that support a compile time feature.